ros/src/tl_detector/tl_detector.py

```diff
@@ -94,7 +94,12 @@ def image_cb(self, msg):
         """
         if not self.waypoints_2d or not self.waypoint_tree:
             return
-        
+
+        t = rospy.get_rostime()
+#        rospy.logwarn("nsec={} mod={}".format(int(t.nsecs * 1e-8), int(t.nsecs * 1e-8) % 3))
+        if int(t.nsecs * 1e-8) % 5 != 0:
+            return
```

Comment:
Image message type has also a header with sequence id. I would rather use that for getting every second image:
`if (msg.header.seq % 2 != 0):`

Reply:
Ups, I misunderstood. You wanted to have 2 images per second, and not every second image. In this case, you have to use a higher value of course. If camera delivers 30 fps, we need every 15th image to have 2 images per second:
`if (msg.header.seq % 15 != 0):`

Reply:
Sorry, this one (finally):
`if (msg.header.seq % 15 == 0):`

Reply:
Did some experiments and realized that we don't know the camera frame rate. It can be different and it seems on the simulator it is different based on system performance. So I'd keep time logic for now.